Revert "zero out auth structs before transfer" which breaks http/https auth.

See:
https://bugs.gentoo.org/show_bug.cgi?id=438718
https://sourceforge.net/tracker/?func=detail&aid=3577557&group_id=976&atid=100976

--
diff --git a/lib/transfer.c b/lib/transfer.c
index 2ad5fad..a00469e 100644
--- a/lib/transfer.c
+++ b/lib/transfer.c
@@ -1433,10 +1433,6 @@ CURLcode Curl_pretransfer(struct SessionHandle *data)
 
   data->state.ssl_connect_retry = FALSE;
 
-  /* zero out auth state */
-  memset(&data->state.authhost, 0, sizeof(struct auth));
-  memset(&data->state.authproxy, 0, sizeof(struct auth));
-
   data->state.authproblem = FALSE;
   data->state.authhost.want = data->set.httpauth;
   data->state.authproxy.want = data->set.proxyauth;
